contracts/extensions/StreamingPayments.sol

```diff
@@ -0,0 +1,414 @@
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./ColonyExtensionMeta.sol";
+
+// ignore-file-swc-108
+
+
+contract StreamingPayments is ColonyExtensionMeta {
+
+  // Events
+
+  event StreamingPaymentCreated(address agent, uint256 streamingPaymentId);
+  event StreamingPaymentClaimed(address agent, uint256 indexed streamingPaymentId, address indexed token, uint256 amount);
+  event PaymentTokenUpdated(address agent, uint256 indexed streamingPaymentId, address indexed token, uint256 amount);
+
+  // Constants
+
+  uint256 constant SLOT = 0;
+  ColonyDataTypes.ColonyRole constant ADMINISTRATION = ColonyDataTypes.ColonyRole.Administration;
+  ColonyDataTypes.ColonyRole constant FUNDING = ColonyDataTypes.ColonyRole.Funding;
+
+  // Storage
+
+  struct StreamingPayment {
+    address payable recipient;
+    uint256 domainId;
+    uint256 startTime;
+    uint256 endTime;
+    uint256 interval;
+  }
+
+  struct PaymentToken {
+    uint256 amount;
+    uint256 lastClaimed;
+  }
+
+  uint256 numStreamingPayments;
+  mapping (uint256 => StreamingPayment) streamingPayments;
+  mapping (uint256 => mapping (address => PaymentToken)) paymentTokens;
+
+  // Modifiers
+
+  modifier validateFundingPermission(uint256 _permissionDomainId, uint256 _childSkillIndex, uint256 _domainId) {
+    require(
+      colony.hasInheritedUserRole(msgSender(), _permissionDomainId, FUNDING, _childSkillIndex, _domainId),
+      "streaming-payments-funding-not-authorized"
+    );
+    _;
+  }
+
+  modifier validateAdministrationPermission(uint256 _permissionDomainId, uint256 _childSkillIndex, uint256 _domainId) {
+    require(
+      colony.hasInheritedUserRole(msgSender(), _permissionDomainId, ADMINISTRATION, _childSkillIndex, _domainId),
+      "streaming-payments-admin-not-authorized"
+    );
+    _;
+  }
+
+  // Public
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("StreamingPayments");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
```

Comment:
Shouldn't... some stuff be disabled on deprecation? Probably worth talking to product about this if you haven't already, I can see arguments for and against allowing editing of already existing payments.

Reply:
Great point, totally slipped by me.